twilio/request_validator.py

```diff
@@ -82,22 +96,29 @@ def validate(self, uri, params, signature):
             params = {}
 
         parsed_uri = urlparse(uri)
+        uri_with_port = uri
+        uri_without_port = uri
+
         if parsed_uri.scheme == "https" and parsed_uri.port:
-            uri = remove_port(parsed_uri)
+            uri_without_port = remove_port(parsed_uri)
+        elif parsed_uri.scheme == "https":
+            uri_with_port = add_port(parsed_uri, 443)
+        elif parsed_uri.scheme == "http" and parsed_uri.port:
+            uri_without_port = remove_port(parsed_uri)
+        elif parsed_uri.scheme == "http":
+            uri_with_port = add_port(parsed_uri, 80)
 
         valid_signature = False  # Default fail
         valid_signature_with_port = False
         valid_body_hash = True  # May not receive body hash, so default succeed
 
-
         query = parse_qs(parsed_uri.query)
         if "bodySHA256" in query and isinstance(params, string_types):
             valid_body_hash = compare(self.compute_hash(params), query["bodySHA256"][0])
-            valid_signature = compare(self.compute_signature(uri, {}), signature)
-            valid_signature_with_port = compare(self.compute_signature(parsed_uri, {}), signature)
+            valid_signature = compare(self.compute_signature(uri_without_port, {}), signature)
```

Comment:
Just set `params = {}` here and then you don't need the same calls in the `else` block.